src/components/data_submitter/ConsentGroupSummary.js

```diff
@@ -0,0 +1,216 @@
+import { div, input, span, p } from 'react-hyperscript-helpers';
+import { isNil } from 'lodash/fp';
+
+
+const primaryConsentText = {
+  generalResearchUse: 'General Research Use',
+  hmb: 'Health/Medical/Biomedical Research Use',
+  diseaseSpecificUse: 'Disease-Specific Research Use',
+  poa: 'Populations, Origins, Ancestry Use',
+  otherPrimary: 'Other',
+};
+
+const secondaryConsentText = {
+  nmds: 'No Methods Development or validation studies (NMDS)',
+  gso: 'Genetic studies only (GSO)',
+  pub: 'Publication Required (PUB)',
+  col: 'Collaboration Required (COL)',
+  irb: 'Ethics Approval Required (IRB)',
+  gs: 'Geographic Restriction (GS-)',
+  mor: 'Publication Moratorium (MOR)',
+  npu: 'Non-profit Use Only (NPU)',
+  otherSecondary: 'Other',
+
+};
+
+// returns { field: string, value: ? } or undefined
+// field is 'set' if it is a boolean and is true
+// or if there is any non-boolean value present.
+const findFirstSetField = (obj, fields) => {
+  for (const i in fields) {
+    const field = fields[i];
+    if (!isNil(obj[field]) && obj[field] !== false) {
+      return field;
+    }
+  }
+  return undefined;
+};
+
+const findAllSetFields = (obj, fields) => {
+  const setFields = [];
+
+  for (const i in fields) {
+    const field = fields[i];
+    if (!isNil(obj[field]) && obj[field] !== false) {
+      setFields.push(field);
+    }
+  }
+  return setFields;
+};
+
+const primaryConsentFields = [
```

Comment:
Would it be worth converting this into like key:value?
```suggestion
const primaryConsentFields = {
   generalResearchUse: 'General Research Use',
   ...
}
```

and then when the array value is required, we can `keys(primaryConsentFields)` to get the array?